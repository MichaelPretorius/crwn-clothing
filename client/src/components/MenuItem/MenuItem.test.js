import React from 'react';
import { shallow } from 'enzyme';

import { MenuItem } from './MenuItem';

describe('MenuItem', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;
  const title = 'hats';
  const size = 'large';
  const imageUrl = 'testimage';

  beforeEach(() => {
    mockMatch = {
      url: '/shop'
    };

    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      size,
      title,
      imageUrl
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  test('should render MenuItem component', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  test('should call history.oush with correct string when MenuItemContainer clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');

    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}shop/${title}`);
  });

  test('should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toEqual(size);
  });

  test('should pass imageUrl to BackgroundImageContainer as the prop imageUrl', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toEqual(imageUrl);
  });
});