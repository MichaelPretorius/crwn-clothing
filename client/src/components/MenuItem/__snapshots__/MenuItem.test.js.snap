// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render MenuItem component 1`] = `
"<MenuItemContainer size=\\"large\\" onClick={[Function: onClick]}>
  <BackgroundImageContainer className=\\"background-image\\" imageUrl=\\"testimage\\" />
  <styled.div className=\\"content\\">
    <styled.div>
      HATS
    </styled.div>
    <styled.div>
      SHOP NOW
    </styled.div>
  </styled.div>
</MenuItemContainer>"
`;
